browser/net/url_context.h

```diff
@@ -102,7 +102,9 @@ struct BraveRequestInfo {
   GURL ipfs_gateway_url;
   bool ipfs_auto_fallback = false;
 
-  bool ShouldMockRequest() const { return !mock_data_url.empty(); }
+  bool ShouldMockRequest() const {
+    return blocked_by == kAdBlocked && !mock_data_url.empty();
```

Comment:
Can we add a test that checks `kAdBlocked` requests are redirected and others are not?
I know that the are rust tests, but we need safety on the browser side, as we keep adding non-obvious logic

Reply:
@iefremov there's currently no support in the adblock engine for parsing `redirect-rule` filters, which is how this should normally come up. But I added a test which loads it from a custom-crafted DAT, which proves it will work correctly when the CRX packager starts shipping out updated files.